apps/api/src/app/widgets/usecases/mark-all-message-as-seen/mark-all-message-as-seen.usecase.ts

```diff
@@ -0,0 +1,34 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { MessageRepository } from '@novu/dal';
+import { AnalyticsService } from '../../../shared/services/analytics/analytics.service';
+import { QueueService } from '../../../shared/services/queue';
+import { ANALYTICS_SERVICE } from '../../../shared/shared.module';
+import { MarkAllMessageAsSeenCommand } from './mark-all-message-as-seen.command';
+
+@Injectable()
+export class MarkAllMessageAsSeen {
+  constructor(
+    private messageRepository: MessageRepository,
+    private queueService: QueueService,
+    @Inject(ANALYTICS_SERVICE) private analyticsService: AnalyticsService
+  ) {}
+
+  async execute(command: MarkAllMessageAsSeenCommand): Promise<number> {
+    const response = await this.messageRepository.markAllUnseenAsSeen(command.subscriberId);
+
+    this.queueService.wsSocketQueue.add({
+      event: 'unseen_count_changed',
+      userId: command.subscriberId,
+      payload: {
+        unseenCount: 0,
+      },
+    });
+
+    this.analyticsService.track('Mark all message as seen - [Notification Center]', command.organizationId, {
+      _subscriber: command.subscriberId,
```

Comment:
Could you remove the `subscriberId` from here? we don't track user defined properties such as the internal userId 🚀

Reply:
Okay